dwave/optimization/src/nodes/_state.hpp

```diff
@@ -173,6 +176,20 @@ class ArrayStateData {
         return size_ - previous_size_;
     }
 
+    bool trim_to(ssize_t new_size) {
```

Comment:
I've been trying to emulate C++ container methods for the most part. So I'd rather add a more generic `resize()` method here.

Reply:
Makes sense, but I'm also not sure we want a `resize()` with full functionality... would we want it to fill in with 0 updates when increasing the size?

Reply:
I'm inclined to say yes. Though obviously developers should be aware of that they are doing. There might be cases where that is sensible :shrug: